lib/ansible/modules/packaging/language/npm.py

```diff
@@ -210,13 +229,13 @@ def uninstall(self):
 
     def list_outdated(self):
         outdated = list()
-        data = self._exec(['outdated'], True, False)
-        for dep in data.splitlines():
-            if dep:
-                # node.js v0.10.22 changed the `npm outdated` module separator
-                # from "@" to " ". Split on both for backwards compatibility.
-                pkg, other = re.split(r'\s|@', dep, 1)
-                outdated.append(pkg)
+        output = self._exec(['outdated', '--json'], True, False)
```

Comment:
I checked that `--json` switch is supported by npm 4.2.0.

Reply:
`--json` is supported only since https://github.com/npm/cli/commit/0b6c9b7c8c5579f4d7d37a0c24d9b7a12ccbe5fe#diff-3d20499d58f14c6f1edfe93d8ba8a8a2. I will add changed for backward compat.

Reply:
I experimented adding test for npm < 1.2.8 with node 0.8.16, but my conclusion is that it should not be necessary to support npm < 1.2.8 to merge the PR.

Last npm < 1.2.8 was shipped with node 0.8 which is not supported anymore. When I tried npm install with 1.2.0, it failed because it had not the correct SSL certificates for the official npm registry...